lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb

```diff
@@ -7,14 +7,20 @@ module OOStructuredFSM
     # Class responsible for handling Repetition state.
     class RepetitionState < OOStructuredState
 
+      MAP_STATE = {
+        StateMachine::METHOD_STATE => "method",
+        StateMachine::CONSTRUCTOR_STATE => "function",
```

Comment:
The value of this hashes (lines 12 and 13) are switched, StateMachine::CONSTRUCTOR_STATE should be "constructor" and StateMachine::GLOBAL_FUNCTION_STATE should be "function", you should also make a test that checks this scenario.